stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/dto/UnreadDtos.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2014-2025 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.api2.model.dto
+
+import com.squareup.moshi.JsonClass
+import java.util.Date
+
+@JsonClass(generateAdapter = true)
+internal data class UnreadDto(
+    val total_unread_count: Int = 0,
+    val total_unread_threads_count: Int = 0,
+    val total_unread_count_by_team: Map<String, Int> = emptyMap(),
+    val channels: List<UnreadChannelDto> = emptyList(),
+    val threads: List<UnreadThreadDto> = emptyList(),
+    val channel_type: List<UnreadChannelByTypeDto> = emptyList(),
```

Comment:
Are we sure we are not receiving null values here?
Other DTOs are returning us null values on the collections, which could crash the execution of our SDK.
If that is the case, could we mark it as nullable and properly parse it within our mappers to not allow nullable values in our business model?

Reply:
Actually, there is only one optional attribute, which is `total_unread_count_by_team`:

<img width="516" height="343" alt="Image" src="https://github.com/user-attachments/assets/0caca235-ca58-43be-b383-2fbfda12eb14" />

Everything else is mandatory.

Swift also follows that:

<img width="549" height="385" alt="Image" src="https://github.com/user-attachments/assets/4b2733b4-636e-4bcd-ae4f-ba01ae9cf486" />
I will update `total_unread_count_by_team`